implementations/rust/ockam/ockam_executor/src/channel.rs

```diff
@@ -50,28 +57,21 @@ struct Inner<T> {
     is_sender_closed: AtomicBool,
 }
 
+impl<T> Inner<T> {
+    fn _len(&self) -> usize {
+        self._length
+    }
+}
+
+/// Sender
 pub struct Sender<T>(Arc<Inner<T>>);
 
 impl<T: core::fmt::Debug> Sender<T> {
     pub async fn send(&self, value: T) -> Result<(), error::SendError<T>> {
```

Comment:
Since `SendFuture` is public, this should probably be a non-async fn that returns `SendFuture<'_, T>` directly. That is, this is equivalent but would be more flexible:

```rs
pub fn send(&self, value: T) -> SendFuture<'_, T> {
    Self {
        inner: &self.0,
        value: Some(value),
    }
}
```

That way, the user could put the `SendFuture` in a struct or something (and it also with gives a reason for having it as a separate type)

Reply:
I agree! 

The reason I'm being weird here is because this a drop-in for `tokio::sync::mpsc::channel` which means our code is expecting a `Result` and not the future. 

Having it as a separate type was an artifact of making it easier to try different approaches to implementing `poll()`

Reply:
Ah, hmmmm. Alright then.